index.js

```diff
@@ -24,18 +24,20 @@ function MockDate(param) {
     if (param instanceof MockDate) {
       this.d = new _Date(param.d);
     } else if (typeof param === 'string') {
-      if (param.match(date_iso_8601_regex) || param.match(date_with_offset) || param.match(date_rfc_2822_regex) || param === '') {
+      if (param.match(date_iso_8601_regex) || param.match(date_with_offset) || param.match(date_rfc_2822_regex)) {
         this.d = new _Date(param);
-      } else if (param.match(local_date_regex)) {
+      } else if (param === '') {
+        this.d = new _Date(param);
+        this.invalid = true;
```

Comment:
This state prop helps the detection below be as simple as possible.

Reply:
I don't think the "invalid" state is actually immutable after construction.  E.g. `new Date('').setTime(12345)` should return a valid date.  Maybe just use `isNaN(this.d.getTime())` below and let the internal date handle tracking invalid?

Reply:
I think you're right. Crazy JavaScript ;) ok will fix.

Reply:
Updated.